File: android/app/src/main/java/com/example/myapplication/main/ui/dashboard/DashboardViewModel.java
package com.example.myapplication.main.ui.dashboard;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.myapplication.main.ui.dashboard.model.CounterModel;

public class DashboardViewModel extends ViewModel {

    private final MutableLiveData<CounterModel> counterModelLiveData = new MutableLiveData<>(new CounterModel(0));

    public LiveData<CounterModel> getCounterModelLiveData() {
        return counterModelLiveData;
    }

    public void processIntent(CounterIntent intent) {
        CounterModel currentModel = counterModelLiveData.getValue();
        CounterModel newModel = null;

        if (intent instanceof CounterIntent.Increment) {
            newModel = new CounterModel(currentModel.getCount() + 1);
        } else if (intent instanceof CounterIntent.Decrement) {
            newModel = new CounterModel(currentModel.getCount() - 1);
        }

        if (newModel != null) {
            counterModelLiveData.setValue(newModel);
        }
    }
}

File: android/app/src/main/java/com/example/myapplication/main/ui/dashboard/DashboardViewModel.kt
package com.example.myapplication.main.ui.dashboard

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.myapplication.main.ui.dashboard.CounterIntent.Decrement
import com.example.myapplication.main.ui.dashboard.model.CounterModel

class DashboardViewModel : ViewModel() {
    val counterModelLiveData = MutableLiveData(CounterModel(0))
    fun getCounterModelLiveData(): LiveData<CounterModel> {
        return counterModelLiveData
    }

    fun processIntent(intent: CounterIntent?) {
        val currentModel = counterModelLiveData.getValue()
        var newModel: CounterModel? = null
        if (intent is CounterIntent.Increment) {
            newModel = CounterModel(currentModel!!.count + 1)
        } else if (intent is Decrement) {
            newModel = CounterModel(currentModel!!.count - 1)
        }
        if (newModel != null) {
            counterModelLiveData.value = newModel
        }
    }
}